django_auth0/models.py

```diff
@@ -1 +1,388 @@
+            acc = APPLICATION.accounts.get(data.get('href'))
+            # materialize it
+            acc.email
+
+            acc = self._mirror_data_from_db_user(acc, data)
+            acc.save()
+            self._save_sp_group_memberships(acc)
+            return acc
+        except StormpathError as e:
+            if e.status == 404:
+                raise self.DoesNotExist('Could not find Stormpath User.')
+            else:
+                raise e
+        finally:
+            self._remove_raw_password()
+
+    def get_full_name(self):
+        return "%s %s" % (self.given_name, self.surname)
+
+    def get_short_name(self):
+        return self.email
+
+    def __unicode__(self):
+        return self.get_full_name()
+
+    def _update_for_db_and_stormpath(self, *args, **kwargs):
+        try:
+            with transaction.atomic():
+                super(Auth0BaseUser, self).save(*args, **kwargs)
+                self._update_stormpath_user(model_to_dict(self), self._get_raw_password())
+        except StormpathError:
+            raise
+        except ObjectDoesNotExist:
+            self.delete()
+            raise
+        except Exception:
+            raise
+
+    def _create_for_db_and_stormpath(self, *args, **kwargs):
+        try:
+            with transaction.atomic():
+                super(Auth0BaseUser, self).save(*args, **kwargs)
+                account = self._create_stormpath_user(model_to_dict(self), self._get_raw_password())
+                self.href = account.href
+                self.username = account.username
+                self.save(*args, **kwargs)
+        except StormpathError:
+            raise
+        except Exception:
+            # we're not sure if we have a href yet, hence we
+            # filter by email
+            accounts = APPLICATION.accounts.search({'email': self.email})
+            if accounts:
+                accounts[0].delete()
+            raise
+
+    def _save_db_only(self, *args, **kwargs):
+        super(Auth0BaseUser, self).save(*args, **kwargs)
+
+    def _remove_raw_password(self):
+        """We need to send a raw password to Stormpath. After an Account is saved on Stormpath
+        we need to remove the raw password field from the local object"""
+
+        try:
+            del self.raw_password
+        except AttributeError:
+            pass
+
+    def _get_raw_password(self):
+        try:
+            return self.raw_password
+        except AttributeError:
+            return None
+
+    def set_password(self, raw_password):
+        """We don't want to keep passwords locally"""
+        self.set_unusable_password()
+        self.raw_password = raw_password
+
+    def check_password(self, raw_password):
+        try:
+            acc = APPLICATION.authenticate_account(self.username, raw_password)
+            return acc is not None
+        except StormpathError as e:
+            # explicity check to see if password is incorrect
+            if e.code == 7100:
```

Comment:
I think these kind of magic literals (numbers, string should be defined as constants)